var/spack/repos/builtin/packages/sina/package.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+from spack import *
+
+
+class Sina(CachedCMakePackage):
+    """Sina C++ Library"""
+
+    homepage = 'https://github.com/LLNL/Sina'
+    url = 'https://github.com/LLNL/Sina/releases/download/v1.10.0/sina-cpp-1.10.0.tar.gz'
+
+    version('develop', git='git@github.com:LLNL/Sina.git',
+            submodules=True, branch='develop')
+    version('1.10.0', sha256='b34379ce8cc5eca5a0f16893053fac75be14c2109d1beed4c6d48e11f9b281c7')
+
+    variant('docs', default=False,
+            description='Allow generating documentation')
+    variant('adiak', default=False,
+            description='Create interface for calling Sina through Adiak')
+    # Availability of the adiak version with optional mpi varies
+    variant('adiak-no-mpi', default=False,
+            description='Create interface for calling Sina through Adiak')
+    variant('test', default=False,
+            description='Build tests')
+
+    depends_on('cmake@3.8.0:', type='build')
+    depends_on('adiak', when='+adiak')
+    depends_on('mpi', when='+adiak')
+    depends_on('adiak ~mpi', when='+adiak-no-mpi')
+    depends_on('doxygen', type='build', when='+docs')
+    depends_on('conduit ~python ~mpi ~hdf5 ~silo ~shared ~test ~hdf5_compat')
+
+    def configure_args(self):
+        spec = self.spec if self.spec is not None else ''
+        return [
+            '-DSINA_BUILD_ADIAK_BINDINGS={0}'.format('YES' if '+adiak' in spec or '+adiak-no-mpi' in spec else 'NO'),
+            '-DSINA_BUILD_TESTS={0}'.format('YES' if '+test' in spec else 'NO'),
+            '-DSINA_BUILD_DOCS={0}'.format('YES' if '+docs' in spec else 'NO'),
+        ]
+
+    def initconfig_package_entries(self):
+        entries = [
+           '#' + 78 * '-',
+           '# Library Dependencies',
+           '#' + 78 * '-'
+        ]
+        path_replacements = {}
+
+        # Try to find the common prefix of the TPL directory, including the compiler
+        # If found, we will use this in the TPL paths
+        compiler_str = str(self.spec.compiler).replace('@','-')
+        prefix_paths = prefix.split(compiler_str)
+        if len(prefix_paths) == 2:
+            tpl_root = os.path.join( prefix_paths[0], compiler_str )
+            path_replacements[tpl_root] = '${TPL_ROOT}'
+            entries.append(cmake_cache_path('TPL_ROOT', tpl_root))
```

Comment:
Is this pointing at some LLNL-specific TPL root?  If so this doesn't belong in an upstream package -- you should use externals for these roots...

Reply:
Hmm.... I don't remember how this works.

We are using BLT and uberenv to build our library. This file is adapted from there. I'll have to figure out how this all works again.

I think though that this is just creating host-config files, and defining a variable named `TPL_ROOT` in them to be able to figure out where the third-party libraries are.

If this is the case (using this to create host-configs), should I just drop this from this package file?

Reply:
I did some looking back, and this is to have a variable named 'TPL_ROOT' in the host-config files. Everything that is relative to this is changed to use this variable. This make it easier to spot the commonalities. For the public build, this is not important to have in the cache file. I will remove it.